File: javascript/grade-school/grade-school.js
function School() {
  this.students = {};
}

School.prototype.roster = function roster() {
  return this.students;
};

School.prototype.grade = function grade(value) {
  return this.students[value] || [];
};

School.prototype.add = function add(name, grade) {
  const currentClass = this.students[grade] || [];
  currentClass.push(name);
  this.students[grade] = currentClass.sort();
};

module.exports = School;

File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @roster = Hash.new { |grade, students| grade[students] = [] }
  end

  def students(grade)
    @roster[grade].sort
  end

  def add(name, grade)
    @roster[grade] << name
  end

  def students_by_grade
    @roster.sort.map do |grade, students|
      {
        grade: grade,
        students: students.sort
      }
    end
  end
end

module BookKeeping
  VERSION = 3
end
